climada/test/test_util_calibrate.py

```diff
@@ -0,0 +1,244 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+Integration tests for calibration module
+"""
+
+import unittest
+
+import pandas as pd
+import numpy as np
+import numpy.testing as npt
+from scipy.optimize import NonlinearConstraint
+from sklearn.metrics import mean_squared_error
+from matplotlib.axes import Axes
+
+from climada.entity import ImpactFuncSet, ImpactFunc
+
+from climada.util.calibrate import (
+    Input,
+    ScipyMinimizeOptimizer,
+    BayesianOptimizer,
+    OutputEvaluator,
+    BayesianOptimizerOutputEvaluator,
+)
+
+from climada.util.calibrate.test.test_base import hazard, exposure
+
+
+class TestScipyMinimizeOptimizer(unittest.TestCase):
+    """Test the TestScipyMinimizeOptimizer"""
+
+    def setUp(self) -> None:
+        """Prepare input for optimization"""
+        self.hazard = hazard()
+        self.hazard.frequency = np.ones_like(self.hazard.event_id)
+        self.hazard.date = self.hazard.frequency
+        self.hazard.event_name = ["event"] * len(self.hazard.event_id)
+        self.exposure = exposure()
+        self.events = [10, 1]
+        self.hazard = self.hazard.select(event_id=self.events)
+        self.data = pd.DataFrame(
+            data={"a": [3, 1], "b": [0.2, 0.01]}, index=self.events
+        )
+        self.impact_to_dataframe = lambda impact: impact.impact_at_reg(["a", "b"])
+        self.impact_func_creator = lambda slope: ImpactFuncSet(
+            [
+                ImpactFunc(
+                    intensity=np.array([0, 10]),
+                    mdd=np.array([0, 10 * slope]),
+                    paa=np.ones(2),
+                    id=1,
```

Comment:
Specify a `haz_type` to prevent `ImpactFuncSet` from logging a warning. Can we maybe just give the output of `hazard()` an arbitrary `haz_type` like `"test"` or something similar? Just rename the column in `exposure()` to `"impf_test"` and pass `haz_type="test"` to all calls of `ImpactFunc` in this test file.

Reply:
Set `haz_type=TEST` in all tests, see d4d677770815dbc62d93c0be4f03049831be25fa